docs/book/user-guide/starter-guide/track-ml-models.md

```diff
@@ -0,0 +1,317 @@
+    Annotated[float, "training_acc"],
+]:
+    # This will return the model version specified in the 
+    # @pipeline decorator. In this case, the production version of 
+    # the `iris_classifier` will be returned in this case.
+    model_version = get_step_context().model_version
+    ...
+```
+
+## Using the stages of a model
+
+A models versions can exist in various stages. These are meant to signify their lifecycle state:
+
+* `staging`: This version is staged for production.
+* `production`: This version is running in a production setting.
+* `latest`: The latest version of the model.
+* `archived`: This is archived and no longer relevant. This stage occurs when a model moves out of any other stage.
+
+{% tabs %}
+{% tab title="Python SDK" %}
+```python
+from zenml.model import ModelVersion
+
+# Get latest model version
+model_version = ModelVersion(
+    name="iris_classifier",
+    version="latest"
+)
+
+# Get a model from a version
+model_version = ModelVersion(
+    name="iris_classifier",
+    version="my_version",
+)
+
+# Pass the stage into the version field
+# to get the model by stage
+model_version = ModelVersion(
+    name="iris_classifier",
+    version="staging",
+)
+
+# This will set this version to production
+model_version.set_stage(stage="production", force=True)
+```
+{% endtab %}
+
+{% tab title="CLI" %}
+```shell
+# List staging models
+zenml model version list <MODEL_NAME> --stage staging 
+
+# Update to production
+zenml model version update <MODEL_NAME> <MODEL_VERSIONNAME> -s production 
+```
+{% endtab %}
+{% tab title="Cloud (Dashboard)" %}
+The [ZenML Cloud](https://zenml.io/cloud) dashboard has additional capabilities, that include easily changing the stage:
+
+![ZenML Cloud Transition Model Stages](../../.gitbook/assets/dcp_transition_stage.gif)
+
+{% endtab %}
+{% endtabs %}
+
+## Associating different types of artifacts with a model
+
+A ZenML model supports linking three types of artifacts:
+
+* `Data artifacts`: These is the default artifacts. If nothing is specified, all artifacts are grouped under this category.
+* `Model artifacts`: If there is a physical model artifact like a pickle file or a model neural network weights file, it should be grouped in this category.
+* `Deployment artifacts`: These artifacts are to do with artifacts related to the endpoints and deployments of the models.
+
+In order to tell ZenML which artifact belongs to which type, one must pass in additional configuration to your artifacts:
+
+```python
+from zenml import get_step_context, step
+from zenml.model import DataArtifactConfig, ModelArtifactConfig
+
+@step
+def svc_trainer(
+    X_train: pd.DataFrame,
+    y_train: pd.Series,
+    gamma: float = 0.001,
+) -> Tuple[
+    # This third argument marks this as a Model Artifact
+    Annotated[ClassifierMixin, "trained_model", ModelArtifactConfig()],
+    # This third argument marks this as a Data Artifact
+    Annotated[float, "training_acc", DataArtifactConfig()],
```

Comment:
As per [this comment](https://github.com/zenml-io/zenml/pull/2111#discussion_r1432439924) from @avishniakov, this needs to use `ArtifactConfig` object instead of the custom `DataArtifactConfig` and `ModelArtifactConfig` since the syntax was updated/changed.